File: Laborator8/Laborator8_Servicii/app/src/main/java/com/example/mp3/MainActivity.java
package com.example.mp3;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button btnPlay, btnStop;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnPlay = findViewById(R.id.btnPlay);
        btnStop = findViewById(R.id.btnStop);

        Intent serviceIntent = new Intent(this, MusicService.class);

        btnPlay.setOnClickListener(v -> {
            startService(serviceIntent);
        });

        btnStop.setOnClickListener(v -> {
            stopService(serviceIntent);
        });
    }
}

File: Laborator5/Kotlin_RecylerView_1/app/src/main/java/com/example/kotlin_recylerview_1/view/MainActivity.kt
package com.example.kotlin_recylerview_1.view

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.activity.viewModels
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.kotlin_recylerview_1.R
import com.example.kotlin_recylerview_1.viewmodel.ProductViewModel


class MainActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView
    private lateinit var adapter: ProductAdapter

    private val viewModel: ProductViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerView = findViewById(R.id.recyclerViewProducts)
        recyclerView.layoutManager = LinearLayoutManager(this)

        adapter = ProductAdapter(this, listOf())
        recyclerView.adapter = adapter

        // Observe LiveData from ViewModel
        viewModel.productList.observe(this, Observer { products ->
            adapter.updateList(products)
        })
    }
}